docs/sources/flow/reference/components/prometheus.remote_write.md

```diff
@@ -361,3 +361,72 @@ prometheus.remote_write "default" {
 `prometheus.remote_write` uses [snappy](https://en.wikipedia.org/wiki/Snappy_(compression)) for compression.
 
 Any labels that start with `__` will be removed before sending to the endpoint.
+
+## Data retention
+
+The `prometheus.remote_write` component reuses the Prometheus Write-Ahead Log
+(WAL) implementation for resiliency against network outages. The component
```

Comment:
The WAL is also useful in scenarios where the alternative (in-memory buffering) would be prohibitively expensive in terms of memory, such as when the samples per second being collected is high. 

Reading from the WAL allows us to control the pace at which collected metrics are buffered in memory.